### PR TITLE
feat: add 'textarea' component

### PR DESCRIPTION
**Pull request by holderb5**
Created textarea component for Issue: https://github.com/denali-design/denali-ember/issues/6

### CODE REVIEW

File: tests/integration/components/denali-textarea-test.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  setupRenderingTest(hooks);
+
+  test('it renders', async function (assert) {
+    assert.expect(2);
+
+    await render(hbs`<DenaliTextarea value="My Input" />`);
+
+    assert.dom('.input textarea').exists('An textarea is rendered within a wrapper div with class `.input`');
+    assert.dom('.input textarea').isNotDisabled('The textarea is not disabled by default');
+    //assert.dom('.input textarea').hasValue('My Input', 'The textarea has the specified value');
```
Comment (Alex-Aralis) [inline tests/integration/components/denali-textarea-test.js:16]: value must be set on textarea like `<textarea>[value]</textarea>`

File: tests/integration/components/denali-textarea-test.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    assert.expect(0);
+
+    this.set('handleInput', (event) => {
+      assert.equal(event.target.value, 'Random Text', 'the action passed in configured correctly');
+    });
+    await render(hbs`<DenaliTextarea
+      {{on "textarea" (action this.handleInput)}}
+    />`);
+
+    /*await fillIn('.input textarea', 'Random Text');
```
Comment (Alex-Aralis) [inline tests/integration/components/denali-textarea-test.js:67]: Didn't import fillIn above.

File: tests/integration/components/denali-textarea-test.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    />`);
+
+    await fillIn('.input textarea', 'Random Text');
+
+    this.set('handleInput', (event) => {
+      assert.equal(event.target.value, 'More Random Text', 'the action is configured correctly');
+    });
+
+    this.element.querySelector('.input textarea').value = 'More Random Text';
+    this.element.querySelector('.input textarea').dispatchEvent(new Event('input'));
```
Comment (Alex-Aralis) [inline tests/integration/components/denali-textarea-test.js:74]: fillin

File: addon/components/denali-textarea.hbs [hunk 4]
```diff
@@ -0,0 +1,10 @@
+{{! Copyright 2020, Verizon Media. Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms. }}
+<div class="input
+  {{this.activeClass}}
+  {{this.warningClass}}"
+>
+  <textarea ...attributes>{{@value}}</textarea>
```
Comment (jkusa) [inline addon/components/denali-textarea.hbs:6]:
```suggestion
  <textarea ...attributes>{{yield}}</textarea>
```

File: tests/dummy/app/templates/denali-textarea.hbs [hunk 5]
```diff
@@ -0,0 +1,8 @@
+<h2>Simple Input</h2>
```
Comment (jkusa) [inline tests/dummy/app/templates/denali-textarea.hbs:1]:
```suggestion
<h2>DenaliTextArea</h2>
```

File: tests/dummy/app/templates/denali-textarea.hbs [hunk 6]
```diff
@@ -0,0 +1,8 @@
...
+<DenaliTextarea />
+<br>
+
+<h2>Input States</h2>
+<DenaliTextarea disabled={{true}} placeholder="Disabled Input" />
```
Comment (jkusa) [inline tests/dummy/app/templates/denali-textarea.hbs:6]: Can we rename this to: `DenaliTextArea`

File: addon/components/denali-textarea.hbs [hunk 7]
```diff
@@ -0,0 +1,10 @@
...
+  {{#if this.warningClass}}
+    <p class="message">{{this.warningMsg}}</p>
```
Comment (jkusa) [inline addon/components/denali-textarea.hbs:8]: Won't this render when there is no message given?

File: tests/integration/components/denali-textarea-test.js [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+
+  test('actions', async function (assert) {
+    assert.expect(2);
+
+    this.set('handleInput', (event) => {
+      assert.equal(event.target.value, 'Random Text', 'the action passed in configured correctly');
+    });
+    await render(hbs`<DenaliTextarea
+      {{on 'input' (action this.handleInput)}}
+    />`);
```
Comment (jkusa) [inline tests/integration/components/denali-textarea-test.js:65]:
```suggestion
    await render(hbs`<DenaliTextarea {{on 'input' this.handleInput}} />`);
```

File: tests/integration/components/denali-textarea-test.js (hunk 3 above)
Comment (jkusa) [reply tests/integration/components/denali-textarea-test.js:74]: Should be able to remove the `settled` with this change

Comment (jkusa) [reply tests/integration/components/denali-textarea-test.js:74]: Also, would you mind making a separate commit to make the same changes in `DenaliInput`?

File: addon/components/denali-textarea.hbs (hunk 4 above)
Comment (holderb5) [reply addon/components/denali-textarea.hbs:6]: Just changing @value to yield causes the tests on values to fail. Do you have other changes you would suggest to resolve that? or should I keep it as @value?

Comment (Alex-Aralis) [reply addon/components/denali-textarea.hbs:6]: My suggestion is to make your own pr fix commit and then push that once all the tests are complete and passing. Don't bother with accepting individual commit suggestions here on github unless they are trivial.

File: addon/components/denali-textarea.hbs (hunk 7 above)
Comment (holderb5) [reply addon/components/denali-textarea.hbs]: Screen Shot 2020-06-17 at 12 36 02 PM
Looking at the dummy site this is what shows for my template. So it doesnt seem to be an issue. They have no <p> within the text area without the warning message.

File: addon/components/denali-textarea.hbs (hunk 4 above)
Comment (Alex-Aralis) [reply addon/components/denali-textarea.hbs]: You gotta test the yield with a snippet like this.

```hbs

<DenaliTextArea>
  Passed Yield Value
</DenaliTextArea>

```

File: tests/integration/components/denali-text-area-test.js [hunk 9]
```diff
@@ -0,0 +1,74 @@
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { fillIn, render } from '@ember/test-helpers';
+import { hbs } from 'ember-cli-htmlbars';
+
+module('Integration | Component | denali-textarea', function (hooks) {
```
Comment (Alex-Aralis) [inline tests/integration/components/denali-text-area-test.js:6]: denali-text-area

File: addon/components/denali-text-area.js [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+  get activeClass() {
+    return this.state === 'active' ? 'is-active' : undefined;
+  }
+
+  get errorClass() {
+    if (this.state === 'error') {
+      if (this.errorMsg) {
+        return 'is-error';
+      }
+      return undefined;
```
Comment (Alex-Aralis) [inline addon/components/denali-text-area.js:24]: double redundant

File: addon/components/denali-text-area.js [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+  @arg(oneOf(states)) state;
+  @arg(string) errorMsg;
+
+  get activeClass() {
+    return this.state === 'active' ? 'is-active' : undefined;
+  }
+
+  get errorClass() {
+    if (this.state === 'error') {
+      if (this.errorMsg) {
```
Comment (jkusa) [inline addon/components/denali-text-area.js:21]: Can we make the state be independent of message? You can have the error styles without an error.

File: addon/components/denali-input.hbs [hunk 12]
```diff
@@ -15,7 +15,7 @@
     <span class="d-icon d-{{this.iconBack}} {{this.iconBackClass}}"></span>
   {{/if}}
 
-  {{#if this.warningClass}}
-    <p class="message">{{this.warningMsg}}</p>
+  {{#if this.errorClass}}
```
Comment (jkusa) [inline addon/components/denali-input.hbs:18]:
```suggestion
  {{#if this.errorMsg}}
```

File: addon/components/denali-text-area.js (hunk 11 above)
Comment (holderb5) [reply addon/components/denali-text-area.js:21]: Ok, yeah I can change that

**Pull request merged**